Update willdurand/negotiation requirement for PHP 8 compatibility

nicolengsy opened the pull request:
The [negotiation package](https://github.com/willdurand/Negotiation/releases/tag/3.0.0) version needs to be ^3.0 in order to be PHP 8 compatible.

Commit:
Update willdurand/negotiation 2.0 -> 3.0

mtrajano commented on composer.json:
```diff
@@ -17,7 +17,7 @@
         "php": "^7.2",
         "psr/http-message": "^1.0",
-        "willdurand/negotiation": "^2.0"
+        "willdurand/negotiation": "^3.0"
```
I would make this `"willdurand/negotiation": "^2.0 || ^3.0"` as it should still be compatible with 2.x if there are no breaking changes

mtrajano replied:
Actually this is fine it was only a naming change and they actually recommend upgrading to 3.0 for php 8 compat: https://github.com/willdurand/Negotiation/releases/tag/3.0.0

akrabat commented:
Thanks. I'll look this week.

mtrajano reviewed (commented):
Can also upgrade the travis file to test on 7.4 and 8.0

Commit:
Add travis tests for 7.4, nightly

nicolengsy commented:
Hi @akrabat! Any updates on this PR?

akrabat commented:
When I change the `require`  `"php": "^7.2",` to `"php": "^7.2 || ^8.0"`, all the tests fail, so I need to understand why and update appropriately.

akrabat committed:
Support PHP 8

Use Laminas/laminas-diactoros and update code for PHP 8 deprecations and
warnings.

akrabat commented:
Hmm... I think I need the GH action on master for it to be picked up.

akrabat committed:
Merge branch 'master' into willdurand-negotiation

akrabat committed:
Update CI: Remove 7.2, add 7.3 & 8.0

akrabat committed:
Remove Negotiator class

This is no longer needed as getOrderedElements() is available upstream.

akrabat committed:
Use willdurrand/negotiation 3.0+

The API is essentially the same as 2.0, but it has getOrderedElements()
in AbstractNegotiator, so we don't need to maintain our own copy.

akrabat commented:
Now it works!

akrabat commented:
@nicolengsy [version 2.0](https://github.com/akrabat/rka-content-type-renderer/releases/tag/2.0) is now released.

nicolengsy commented:
@akrabat Thank you so much!

ingluisjimenez commented:
thank you for merging and releasing @akrabat ❤️
